internal/util/sizeof_test.go

```diff
@@ -0,0 +1,376 @@
+			expected_size: sizeOfEmptyStringArray,
+		},
+		{
+			t:             map[string]int{"few ": 432, "fewfgwe": -21, "": 1, "fewawef": 0},
+			expected_size: sizeOfEmptyStringIntMap,
+		},
+	} {
+		calculatedSize := UnsafeSizeOf(&tc.t)
+		AssertEq(tc.expected_size, calculatedSize)
+	}
+}
+
+func (t *SizeofTest) TestContentSizeOfString() {
+	for _, tc := range []struct {
+		str                   string
+		expected_content_size int
+	}{
+		{
+			str:                   "",
+			expected_content_size: 0,
+		},
+		{
+			str:                   "hello",
+			expected_content_size: 5,
+		},
+		{
+			str:                   "hello-world",
+			expected_content_size: 11,
+		},
+	} {
+		AssertEq(tc.expected_content_size, contentSizeOfString(&tc.str))
+		AssertEq(tc.expected_content_size+sizeOfEmptyString, nestedSizeOfString(&tc.str))
+	}
+}
+
+func (t *SizeofTest) TestNestedContentSizeOfArrayOfStrings() {
+	for _, tc := range []struct {
+		strs                         []string
+		expected_nested_content_size int
+	}{
+		{
+			strs:                         []string{},
+			expected_nested_content_size: 0,
+		},
+		{
+			strs:                         []string{""},
+			expected_nested_content_size: sizeOfEmptyString,
+		},
+		{
+			strs:                         []string{"", ""},
+			expected_nested_content_size: 2 * sizeOfEmptyString,
+		},
+		{
+			strs:                         []string{"hello", ""},
+			expected_nested_content_size: 2*sizeOfEmptyString + 5,
+		},
+		{
+			strs:                         []string{"hello", "hello-world"},
+			expected_nested_content_size: 2*sizeOfEmptyString + 5 + 11,
+		},
+	} {
+		AssertEq(tc.expected_nested_content_size, nestedContentSizeOfArrayOfStrings(&tc.strs))
+		AssertEq(tc.expected_nested_content_size+sizeOfEmptyStringArray, nestedSizeOfArrayOfStrings(&tc.strs))
+	}
+}
+
+func (t *SizeofTest) TestNestedContentSizeOfStringToStringMap() {
+	for _, tc := range []struct {
+		m                            map[string]string
+		expected_nested_content_size int
+	}{
+		{
+			m:                            map[string]string{},
+			expected_nested_content_size: 0,
+		},
+		{
+			m:                            map[string]string{"hello": "to you"},
+			expected_nested_content_size: sizeOfEmptyString + 5 + sizeOfEmptyString + 6,
+		},
+		{
+			m:                            map[string]string{"a": ""},
+			expected_nested_content_size: sizeOfEmptyString + 1 + sizeOfEmptyString,
+		},
+		{
+			m:                            map[string]string{"": ":"},
+			expected_nested_content_size: sizeOfEmptyString + sizeOfEmptyString + 1,
+		},
+		{
+			m:                            map[string]string{"a": "b1", "xyz": "alpha"},
+			expected_nested_content_size: sizeOfEmptyString + 1 + sizeOfEmptyString + 2 + sizeOfEmptyString + 3 + sizeOfEmptyString + 5,
+		},
+	} {
+		AssertEq(tc.expected_nested_content_size, nestedContentSizeOfStringToStringMap(&tc.m))
+	}
+}
+
+func (t *SizeofTest) TestNestedContentSizeOfStringToStringArrayMap() {
```

Comment:
We should take a case, whose value is an array of string of size 2 or more than 2.

Reply:
Done. Updated the last test case.